byway/src/main/webapp/utils.js

```diff
@@ -0,0 +1,32 @@
+/** Script that contains shared functions and variables*/
+
+// map object used in the route page
+let map; 
+
+/**
+ * Uses placeId to retrieve details like coordinates, place name, etc.
+ * @param {String} placeId a textual identifier that uniquely identifies a place
+ * @return {Promise} result a Place Result object with fields name, geometry, id, etc.
+ */
+function findPlace(placeId) {
+  console.log(placeId);
+  const request = {
+    placeId: placeId,
+    fields: ['name', 'geometry', 'place_id', 'photos', 'formatted_address']
+  }
+  const result = new Promise((resolve, reject) => {
+    placesService.getDetails(request, (result, status) => {
```

Comment:
Since placesService is not defined in this file, it probably makes better sense to pass it as a parameter rather than rely on it being a global variable.

Reply:
I changed it, but what's wrong with it being a global variable? Is it more error prone this way?